anise-cli/src/main.rs

```diff
@@ -331,3 +253,58 @@ fn main() -> Result<(), CliErrors> {
         }
     }
 }
+
+fn truncate_daf_by_id<R>(
+    args::TruncateById {
+        input,
+        output,
+        id,
+        start,
+        end,
+    }: args::TruncateById,
+    bytes: Bytes,
+) -> Result<(), CliErrors>
+where
+    R: NAIFSummaryRecord,
+{
+    info!("Loading {input:?} as DAF/PCK");
+    let fmt = DAF::<R>::parse(bytes).context(CliDAFSnafu)?;
+
+    let mut ids = HashSet::new();
+    for summary in fmt.data_summaries().context(CliDAFSnafu)? {
+        ids.insert(summary.id());
+    }
+
+    info!("IDs present in file: {ids:?}");
+
+    let (summary, idx) = fmt.summary_from_id(id).context(CliDAFSnafu)?;
+
+    let data_type = summary.data_type().map_err(|err| CliErrors::CliDataType {
+        error: Box::new(err),
+    })?;
+    ensure!(
+                        data_type == DafDataType::Type2ChebyshevTriplet,
+                        ArgumentSnafu {
+                            arg: format!("{input:?} is of type {data_type:?}, but operation is only valid for Type2ChebyshevTriplet")
+                        }
+                    );
+
+    let segment = fmt.nth_data::<Type2ChebyshevSet>(idx).unwrap();
```

Comment:
These unwrap calls should probably bubble up.

Reply:
Yep sounds good I can add to the error list